samples/WinForms/Dx12Panel.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright © Tanner Gooding and Contributors. Licensed under the MIT License (MIT). See License.md in the repository root for more information.
+
+using System;
+using System.Windows.Forms;
+using TerraFX.Samples.DirectX.D3D12;
+namespace TerraFX.Samples.WinForms
+{
+    public class DX12Panel : Panel
+    {
+        private readonly DX12Viewport _dxVp;
+        public DX12Viewport DxVp => _dxVp;
+
+        public DX12Panel(int width, int height) : base()
+        {
+            _dxVp = InitializeViewport(width, height);
+            Resize += OnResize;
+        }
+
+        private DX12Viewport InitializeViewport(int width, int height)
+        {
+            var dxVp = new DX12Viewport((uint)width, (uint)height, Handle);
```

Comment:
`width` and `height` should be validated to be non-negative if this is being done, otherwise users will pass in a value and it may wrap around.

Reply:
assuming this is an appropriate check and response:
            `if (width <= 0 || height <= 0) throw new ArgumentException("width and height must be positive and non-zero");`
Let me know if you had something else in mind.

Reply:
If https://github.com/terrafx/terrafx.interop.windows/pull/52#discussion_r421548758 is taken care of, then `Panel` should end up validating these parameters for you.